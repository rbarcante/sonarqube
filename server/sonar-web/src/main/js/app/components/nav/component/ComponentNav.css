.navbar-context-branches {
  display: inline-flex;
  justify-content: center;
  line-height: calc(2 * var(--gridSize));
  margin-left: calc(2 * var(--gridSize));
  font-size: var(--baseFontSize);
}

.navbar-context-meta-branch-menu-item {
  display: flex !important;
  justify-content: space-between;
  align-items: center;
}

.navbar-context-meta-branch-menu-item-name {
  flex: 1;
  max-width: 300px;
}

.navbar-context-meta-branch-menu-item-actions {
  height: 12px;
  margin-left: 32px;
}
