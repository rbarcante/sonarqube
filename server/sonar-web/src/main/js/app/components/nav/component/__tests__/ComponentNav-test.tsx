/*
 * SonarQube
 * Copyright (C) 2009-2016 SonarSource SA
 * mailto:contact AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
/* eslint-disable import/order */
import * as React from 'react';
import { mount, shallow } from 'enzyme';
import ComponentNav from '../ComponentNav';

jest.mock('../ComponentNavMeta', () => ({
  // eslint-disable-next-line
  default: function ComponentNavMeta() {
    return null;
  }
}));

jest.mock('../ComponentNavHeader', () => ({
  // eslint-disable-next-line
  default: function ComponentNavHeader() {
    return null;
  }
}));

jest.mock('../ComponentNavMenu', () => ({
  // eslint-disable-next-line
  default: function ComponentNavMenu() {
    return null;
  }
}));

jest.mock('../../../RecentHistory', () => ({
  default: { add: jest.fn() }
}));

jest.mock('../../../../../api/ce', () => ({
  getTasksForComponent: jest.fn(() => Promise.resolve({ queue: [] }))
}));

const getTasksForComponent = require('../../../../../api/ce').getTasksForComponent as jest.Mock<
  any
>;

const component = {
  breadcrumbs: [{ key: 'component', name: 'component', qualifier: 'TRK' }],
  key: 'component',
  name: 'component',
  organization: 'org',
  qualifier: 'TRK'
};

it('loads status', () => {
  getTasksForComponent.mockClear();
  mount(<ComponentNav branches={[]} component={component} location={{}} />);
  expect(getTasksForComponent).toBeCalledWith('component');
});

it('renders', () => {
  const wrapper = shallow(<ComponentNav branches={[]} component={component} location={{}} />);
  wrapper.setState({ isInProgress: true, isPending: true });
  expect(wrapper).toMatchSnapshot();
});
