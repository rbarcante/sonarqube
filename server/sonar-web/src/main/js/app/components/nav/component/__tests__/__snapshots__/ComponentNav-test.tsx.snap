// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<ContextNavBar
  height={92}
  id="context-navigation"
  notif={
    <ComponentNavBgTaskNotif
      component={
        Object {
          "breadcrumbs": Array [
            Object {
              "key": "component",
              "name": "component",
              "qualifier": "TRK",
            },
          ],
          "key": "component",
          "name": "component",
          "organization": "org",
          "qualifier": "TRK",
        }
      }
      currentTask={undefined}
      isInProgress={true}
      isPending={true}
    />
  }
>
  <ComponentNavHeader
    branches={Array []}
    component={
      Object {
        "breadcrumbs": Array [
          Object {
            "key": "component",
            "name": "component",
            "qualifier": "TRK",
          },
        ],
        "key": "component",
        "name": "component",
        "organization": "org",
        "qualifier": "TRK",
      }
    }
    location={Object {}}
  />
  <ComponentNavMeta
    component={
      Object {
        "breadcrumbs": Array [
          Object {
            "key": "component",
            "name": "component",
            "qualifier": "TRK",
          },
        ],
        "key": "component",
        "name": "component",
        "organization": "org",
        "qualifier": "TRK",
      }
    }
  />
  <ComponentNavMenu
    component={
      Object {
        "breadcrumbs": Array [
          Object {
            "key": "component",
            "name": "component",
            "qualifier": "TRK",
          },
        ],
        "key": "component",
        "name": "component",
        "organization": "org",
        "qualifier": "TRK",
      }
    }
    location={Object {}}
  />
</ContextNavBar>
`;
